### PR TITLE
feat(react): add `TabPanel` component

### PR DESCRIPTION
**Pull request by JayaShakthi97**
### Purpose

> $subject

### Related Issues
- https://github.com/wso2/oxygen-ui/issues/2

### Related PRs
- N/A

### Checklist
- [ ] UX/UI review done on the final implementation.
- [ ] Story provided. (Add screenshots)
- [ ] Manual test round performed and verified.
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Documentation provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: packages/react/src/components/TabPanel/TabPanel.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+# TabPanel
+
+- [Overview](#overview)
+- [Props](#props)
+- [Usage](#usage)
+
+## Overview
+
+TabPanel component can be used with Tabs component to implement navigation between different views or sections.
```
Comment (savindi7) [inline packages/react/src/components/TabPanel/TabPanel.stories.mdx:23]: Shall we have a different `Overview` description? as the `Tab` component's one is similar to this.

File: packages/react/src/components/TabPanel/TabPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import clsx from 'clsx';
+import {forwardRef, ForwardRefExoticComponent, MutableRefObject, ReactElement} from 'react';
+import {WithWrapperProps} from '../../models';
+import {composeComponentDisplayName} from '../../utils';
+import Box from '../Box';
+import './tab-panel.scss';
+
+export interface TabPanelProps extends MuiBoxProps {
+  /*
+   * The value of the `TabPanel`.
```
Comment (savindi7) [inline packages/react/src/components/TabPanel/TabPanel.tsx:29]: is this correct?

File: packages/react/src/components/TabPanel/TabPanel.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+   */
+  index: number;
+  /*
+   * The value of the currently selected `TabPanel`.
```
Comment (savindi7) [inline packages/react/src/components/TabPanel/TabPanel.tsx:33]:
```suggestion
   * The value of the selected `TabPanel`.
```

File: packages/react/src/components/TabPanel/TabPanel.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+const TabPanel: ForwardRefExoticComponent<TabPanelProps> & WithWrapperProps = forwardRef(
+  (props: TabPanelProps, ref: MutableRefObject<HTMLDivElement>): ReactElement => {
+    const {className, children, value, index, ...rest} = props;
+
+    const classes: string = clsx('oxygen-tab-panel', className);
+
+    return (
+      <Box className={classes} ref={ref} role="tabpanel" hidden={value !== index} {...rest}>
+        {value === index && <Box>{children}</Box>}
```
Comment (savindi7) [inline packages/react/src/components/TabPanel/TabPanel.tsx:48]: WDYT about adding a default padding here?

File: packages/react/src/components/TabPanel/TabPanel.tsx (hunk 3 above)
Comment (JayaShakthi97) [reply packages/react/src/components/TabPanel/TabPanel.tsx]: Addressed with 52de18c470e95f639d32394e2bda8233c7ef2833

**Review by savindi7 — APPROVED** [review]

**Pull request merged**